src/markermap/utils.py

```diff
@@ -522,7 +560,31 @@ def __init__(self, input_size, hidden_layer_size, z_size, output_size = None, bi
         self.encoder, self.enc_mean, self.enc_logvar = make_encoder(input_size,
                 hidden_layer_size, z_size, bias = bias, batch_norm = batch_norm)
 
-        self.decoder, self.dec_logvar = make_gaussian_decoder(output_size, hidden_layer_size, z_size, bias = bias, batch_norm = batch_norm)
+        if (decoder is not None and dec_logvar is None) or (decoder is None and dec_logvar is not None):
+            print(
+                'VAE::__init__: WARNING! If decoder is specified, dec_logvar should also be specified, and vice versa'
+            )
+
+        if decoder is None:
+            decoder = make_gaussian_decoder(
+                output_size,
+                hidden_layer_size,
+                z_size,
+                bias = bias,
+                batch_norm = batch_norm,
+            )
+
+        if dec_logvar is None:
+            dec_logvar = make_gaussian_decoder(
+                output_size,
+                hidden_layer_size,
+                z_size,
+                bias = bias,
+                batch_norm = batch_norm,
+            )
+
+        self.decoder = decoder
+        self.dec_logvar = dec_logvar
```

Comment:
@beelze-b This was my solution to not use the `deepcopy`, I think this should be good right? You can pass either decoder and dec_logvar (and if you pass only 1, it will warn you that you should pass both). If they aren't passed, they are set to `make_gaussian_decoder` which should avoid the deepcopy issue. 

What I didn't do was make it so that you can specify the loss function like we were saying we would want to do if you were swapping to the bernoulli decoder. So right now this would really just help you pass different architectures for the gaussian decoder, like more hidden layers or something.

Reply:
This looks fine to me. Weird when I refreshed earlier I was seeing the old version. 

Sorry for the confusion.